src/pages/workspaces/workspace/Tools.js

```diff
@@ -103,16 +134,29 @@ export const Tools = _.flow(
   }
 
   render() {
-    const { namespace, name, listView, viewToggleButtons } = this.props
-    const { loading, configs } = this.state
+    const { namespace, name, listView, viewToggleButtons, workspace: { workspace } } = this.props
+    const { loading, configs, copyingTool, deletingTool } = this.state
     return h(PageFadeBox, [
       div({ style: { display: 'flex', alignItems: 'center', justifyContent: 'space-between', marginBottom: '1rem' } }, [
         div({ style: { ...Style.elements.sectionHeader, textTransform: 'uppercase' } }, ['Tools']),
-        viewToggleButtons
+        viewToggleButtons,
+        copyingTool && h(ExportToolModal, {
+          thisWorkspace: workspace, methodConfig: copyingTool,
+          onDismiss: () => this.setState({ copyingTool: undefined })
+        }),
+        deletingTool && h(DeleteToolModal, {
+          workspace, methodConfig: deletingTool,
+          onDismiss: () => this.setState({ deletingTool: undefined }),
+          onSuccess: () => Nav.goToPath('workspace-tools', _.pick(['namespace', 'name'], workspace))
+        })
       ]),
       div({ style: styles.cardContainer(listView) }, [
         _.map(config => {
-          return h(ToolCard, { key: `${config.namespace}/${config.name}`, namespace, name, config, listView })
+          return h(ToolCard, {
+            onCopy: () => this.setState({ copyingTool: config }),
```

Comment:
We could use a little less state here by just storing the ID (which in this case is probably a compound object with namespace and name), and then looking up the configs when rendering the modals above.